### ISSUE TITLE
Add exact (erf-based) GELU to the ml family

### ISSUE THREAD

**Issue by bravo1goingdark** [themankindproject/lanes#9]
## Summary

Add the exact (erf-based) GELU to the `ml` family. Most crates ship only
the tanh approximation; BERT's original activation is the erf form and
users comparing against PyTorch's default expect it.

Parent: [#2](https://github.com/themankindproject/lanes/issues/2) (Tier 2)

## Proposed API

```rust
// src/algorithms/ml.rs, f32 and f64 submodules
pub fn gelu_exact(values: &[f32]) -> Vec<f32>
```

`0.5 * x * (1 + erf(x / sqrt(2)))`, gated on `alloc`.

## Semantics

Match `torch.nn.GELU(approximate='none')` within the accuracy bound
documented for the erf kernel it builds on. Keep the existing tanh-based
`gelu` untouched (it is the fast path).

## Backend notes

- Lands after the `erf` kernel (special-functions issue); until then, use
  a scalar `erf` reference with a documented ULP bound.
- SIMD path is a direct composition: `mul_add` on top of the erf kernel.

## Acceptance checklist

- [ ] Cross-checked against PyTorch exact GELU on a fixed test vector set
      (document the tolerance).
- [ ] Proptest vs scalar reference; fuzz target.
- [ ] Bench vs existing tanh `gelu`; README + CHANGELOG updated.
